hw/ip/keymgr/doc/_index.md

```diff
@@ -0,0 +1,212 @@
+
+## Life Cycle Connection
+The function of the key manager is directly tied to the life cycle controller.
+During specific life cycle states, the key manager is explicitly disabled.
+
+When disabled, key manager will not advance or generate and if currently in an operational state, will automatically transition to `Disabled`.
+
+
+## Commands in Each State
+During each state, there are 3 valid commands software can issue:
+*  Advance state
+*  Output generation
+*  Identity generation
+
+The software is able to select a command and trigger the key manager FSM to process one of the commands.
+If a command is valid during the current working state, it is processed and acknowledged when complete.
+If a command is invalid, the key manager FSM processes with random, dummy data, but does not update working state or relevant output registers.
+
+For each valid command, a set of inputs are selected and sequenced to the KMAC module.
+
+## Generating Output Key
+The generate output command is composed of 2 options
+*  Generate output key for software, referred to as `generate-output-sw`
+*  Generate output key for hardware, referred to as `generate-output-hw`
+
+The hardware option is meant specifically for symmetric side load use cases.
+When this option is issued, the output of the KMAC invocation is not stored in software visible registers, but instead in hardware registers that directly output to symmetric primitives such as AES, HMAC and KMAC.
+
+## KMAC Operations
+All invoked KMAC operations expect the key in two shares.
+This means the working states, even though functionally 256b, are maintained as 512b.
+
+For advance-state and `generate-output` commands, the KMAC emitted output are also in 2-shares.
+Software is responsible for determining if the key should be preserved in shares or combined, depending on the use case.
+
+
+## Block Diagram
+The following is a high level block diagram of the key manager.
+
+![Key Manager Block Diagram](keymgr_block_diagram.svg)
+
+## Design Details
+
+Key manager is primarily composed of two components:
+*  keymgr_ctrl
+*  keymgr_kmac_if
+
+### Key Manager Control
+
+The key manager control block manages the working state, sideload key updates, as well as what commands are valid in each state.
+It also handles the life cycle keymgr_en input, which helps disable the entire key manager function in the event of an escalation.
+
+![Key Manager Control Block Diagram](keymgr_control_diagram.svg)
+
+
+### KMAC Interface Control
+
+The KMAC interface control represents the bulk of key manager logic.
+Based on input from key manager control, this module selects the inputs for each given command and sequences the data to KMAC.
+
+![Key Manager KMAC Interface Block Diagram](keymgr_kmac_if_diagram.svg)
+
+## Hardware Interfaces
+{{< hwcfg "hw/ip/keymgr/data/keymgr.hjson" >}}
+
+# Programmers Guide
+
+## Initialize
+
+## Advance or Generate
+Software selects a command and triggers a “start” to.
```

Comment:
I may be wrong, but these quotes look like they are non-ASCII (there are other instances further above). consider replacing those with ASCII ones...

Reply:
o man good catch.  Thanks.